Sources/Request Strategies/MLS/Actions/CountSelfMLSKeyPackagesActionHandler.swift

```
@@ -0,0 +1,68 @@
+//
+// Wire
+// Copyright (C) 2022 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+
+class CountSelfMLSKeyPackagesActionHandler: ActionHandler<CountSelfMLSKeyPackagesAction> {
+
+    // MARK: - Methods
+
+    override func request(
+        for action: ActionHandler<CountSelfMLSKeyPackagesAction>.Action,
+        apiVersion: APIVersion) -> ZMTransportRequest? {
+
+            return ZMTransportRequest(
+                path: "/mls/key-package/self/\(action.clientID)/count",
```

We should guard that the client ID is not empty, because it if is, then it would create an invalid path. In which case we'd need to notify the action failure and return nil. Or do you think it's better to do the validation when creating the action?

---

Yes! we should guard `client ID` here and also while creating the action for double check.

---

In addition to returning `nil`, I think it'd be good to also notify a failure on the action's result handler, so that we can surface the error.

---

Yes I have also added failure notification.